x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/TextExpansionQueryBuilder.java

```diff
@@ -51,6 +52,34 @@ public class TextExpansionQueryBuilder extends AbstractQueryBuilder<TextExpansio
     private SetOnce<TextExpansionResults> weightedTokensSupplier;
     private final TokenPruningConfig tokenPruningConfig;
 
+    public enum AllowedFieldType {
+        RANK_FEATURES("rank_features"),
```

Comment:
Question: Instead of hardcoding Strings here would it make sense to reference `RankFeaturesFieldMapper.CONTENT_TYPE` and `SparseVectorFieldMapper.CONTENT_TYPE` here?

Reply:
@kderusso I am trying to import RankFeaturesFieldMapper in TextExpansionQueryBuilder.java

```
import org.elasticsearch.index.mapper.extras.RankFeaturesFieldMapper;
```

However, I am getting Java Compiler error: Cannot resolve symbol 'RankFeaturesFieldMapper'

I am not a Java expert. Do you have any thoughts on what I am doing wrong here? Thank you.

Reply:
I will try to add this in a separate PR.